### PR TITLE
Create issue for detached addons

### PR DESCRIPTION
**Pull request by mdegat01**
## Proposed change

Create an issue if we detect a detached addon so we can make this into a repair and inform users they aren't receiving updates for these addons.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Ruff (`ruff format supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Comment by mdegat01** [comment]
@agners I choose not to include the repo URL in stored addon data. Detached addons where the repo was removed generally shouldn't happen other then current users hit with bugs in the past (and that won't help them).

Each addon includes a URL linking to where to find information for that particular addon. So I'm going to link to that in the repair and tell people to find more info there. That is available even for existing users in this situation.

File: supervisor/resolution/checks/detached_addon.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @property
+    def context(self) -> ContextType:
+        """Return a ContextType enum."""
+        return ContextType.ADDON
+
+    @property
+    def states(self) -> list[CoreState]:
+        """Return a list of valid states when this check can run."""
+        return [CoreState.RUNNING, CoreState.SETUP]
```
Comment (agners) [inline supervisor/resolution/checks/detached_addon.py:46]: Doesn't `RUNNING` makes this a periodical check?

I think once at startup should be enough.

Comment (pvizeli) [inline supervisor/resolution/checks/detached_addon.py:46]:
```suggestion
        return [CoreState.STARTUP, CoreState.RUNNING]
```
At this point, addons are loaded, SETUP can be to early I feel. I guess we need RUNNING as well detect detach add-ons after store updates

Comment (mdegat01) [reply supervisor/resolution/checks/detached_addon.py]: @pvizeli we can do either. Currently the healthcheck during `SETUP` is at the very end after everything else in that step so all addon information is loaded:
https://github.com/home-assistant/supervisor/blob/67ec89c9567016182d77d22968cb9632820f091f/supervisor/core.py#L191

I just removed `RUNNING` from there as per @agners request although I can add it back if we want. Once at startup of supervisor seemed like enough, this should be uncommon other then possibly the source code of a local add-on being removed so I don't think we need to catch this immediately.

**Review by pvizeli — APPROVED** [review]
Okay, let's test that

**Pull request merged**